cmd/snap-confine/seccomp-support.c

```diff
@@ -152,24 +260,27 @@ bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag)
 	// set on the system.
 	validate_bpfpath_is_safe(profile_path);
 
-	/* The extra space has dual purpose. First of all, it is required to detect
-	 * feof() while still being able to correctly read MAX_BPF_SIZE bytes of
-	 * seccomp profile.  In addition, because we treat the profile as a
-	 * quasi-string and use sc_streq(), to compare it. The extra space is used
-	 * as a way to ensure the result is a terminated string (though in practice
-	 * it can contain embedded NULs any earlier position). Note that
-	 * sc_read_seccomp_filter knows about the extra space and ensures that the
-	 * buffer is never empty. */
-	char bpf[MAX_BPF_SIZE + 1] = { 0 };
-	size_t num_read = sc_read_seccomp_filter(profile_path, bpf, sizeof bpf);
-	if (sc_streq(bpf, "@unrestricted\n")) {
+	// workaround bug in gcc from 14.04, the pragma can be removed when
+	// we stop supporting 14.04
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-braces"
+	struct sc_seccomp_file_header hdr = { 0 };
+#pragma GCC diagnostic pop
+	FILE *file SC_CLEANUP(sc_cleanup_file) =
+	    sc_must_read_and_validate_header_from_file(profile_path, &hdr);
+	if (hdr.unrestricted == 0x1) {
```

Comment:
IMO This should be a mask test, not a value comparison.
```suggestion
	if (hdr.unrestricted & 0x1) {
```

Reply:
Will do

Reply:
Done in 2ae52cbfa2d8635f7365cbe8e21ad31dd4a02f57